services/ui-src/src/components/pages/Dashboard/DashboardPage.tsx

```diff
@@ -92,6 +92,9 @@ export const DashboardPage = ({ reportType }: Props) => {
   const [selectedReport, setSelectedReport] = useState<AnyObject | undefined>(
     undefined
   );
+  const [wpToCopyFrom, setWorkPlanToCopyFrom] = useState<boolean>(
```

Comment:
So, there is already a function to set the WP to copy from in the Report Provider. Which makes this state a little redundant.

I was thinking, maybe we could instead move where we currently call `setWorkPlanToCopyFrom` [in the fetch call](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/blob/main/services/ui-src/src/components/reports/ReportProvider.tsx#L116). Like maybe move that line to an `else` block on [line 138](https://github.com/Enterprise-CMCS/macpro-mdct-mfp/blob/main/services/ui-src/src/components/reports/ReportProvider.tsx#L138)?

Reply:
Good catch, I had to look through it before commenting on this but I think the undefined might be causing the issue.  `fetchReportForSarCreation(activeState)` is being called in DashboardPage (line 119) and it's not resolving right away which I think is returning an `undefined` at first which is causing the banner to temporarily flicker. 

I added a null as the default value so even if fetchReportForSarCreation doesn't resolve right away the banner wont render until a null or `workPlanToCopyFrom` object comes through. 

Open to suggestions of this solution though.